src/main/java/edu/kit/ipd/crowdcontrol/objectservice/moneytransfer/MailParser.java

```diff
@@ -0,0 +1,74 @@
+package edu.kit.ipd.crowdcontrol.objectservice.moneytransfer;
+
+import edu.kit.ipd.crowdcontrol.objectservice.database.model.tables.records.GiftCodeRecord;
+
+import javax.mail.BodyPart;
+import javax.mail.Message;
+import javax.mail.MessagingException;
+import javax.mail.Multipart;
+import java.io.*;
+import java.util.regex.Matcher;
+import java.util.regex.Pattern;
+
+/**
+ * Describes a parser to extract amazon giftcodes from mails.
+ */
+public class MailParser {
+
+    protected static GiftCodeRecord parseAmazonGiftCode(Message msg) throws MessagingException, IOException, AmazonMailFormatChangedException{
+        //Extract Message
+        String message;
+        try {
+            Multipart parts = (Multipart) msg.getContent();
+            BodyPart body = parts.getBodyPart(0);
+            Multipart innerMsg = (Multipart) body.getContent();
+            BodyPart textBody = innerMsg.getBodyPart(0);
+            message = textBody.getContent().toString();
+        } catch (ClassCastException e) {
+            throw new AmazonMailFormatChangedException();
+        }
+        //Parse Message
+        String messageStr = message.replaceAll(" ","");
+
+        String codePatternStr = "[0-9A-Z]+(-[0-9A-Z]+)+";
+        Pattern codePattern = Pattern.compile(codePatternStr);
+        Matcher codeMatcher = codePattern.matcher(messageStr);
+
+        String amountPatternStr = "[0-9]+[,|.][0-9][0-9][€]";
```

Comment:
`[,|.]`, why's there a `|` in there? It's a character class, it will match `|`. I don't think that's intended.